Dockerfile

```diff
@@ -4,34 +4,35 @@ FROM golang:alpine as builder
 # queries required to connect to linked containers succeed.
 ENV GODEBUG netdns=cgo
 
-# Install dependencies and build the binaries.
+# Copy in the local repository to build from.
+COPY . /go/src/github.com/lightningnetwork/lnd
```

Comment:
There's a `Dockerfile` here https://github.com/lightningnetwork/lnd/blob/master/docker/lnd/Dockerfile which builds from the local repo.  Is there a reason you prefer to add this change to this `Dockerfile`?

Reply:
The idea is to get rid of the dockerfile you mentionned. Once you have a dockerfile for the local state, then you also have a dockerfile for master (it's only a matter of checking out the right branch and syncing).

Also having a dockerfile for master is pretty useless, since the master branch should not be used in production.

Reply:
By making this change you necessitate requiring the user to checkout and clone this repo, while as it stands this could be used as a stand-alone dockerfile.  I agree it's unlikely that someone will want to use master for "production", but instead will probably pass a [tag](https://github.com/lightningnetwork/lnd/pull/2520/files#diff-dad9d3cc0978f8b909f0876f7b120a68R87).

Ideally, we'd like to be able to just pass a flag to this dockerfile to indicate if we want to build locally or download from GitHub.

Reply:
> By making this change you necessitate requiring the user to checkout and clone this repo

Yes, that's their responsibility to download the code and verify that they got the good release (signed, up to date, etc…). At the end you still need to download the code, inside the dockerfile or elsewhere.

> Ideally, we'd like to be able to just pass a flag to this dockerfile to indicate if we want to build locally or download from GitHub.

It could be a solution if we can't reach a middle ground. However, it would complexify the dockerfile (again, in my opinion, for no practical improvement).

Reply:
Also, if the dockerfile gets updated, a checkout of the tag / commit would get you the "correct" dockerfile for that release.